### PR TITLE
ENH: Add support for full-text queries using syntax `db('keyword')`

### PR DESCRIPTION
**Pull request by danielballan**
Exposing the full-text search added in NSLS-II/metadatastore#215 via the broker. Demo:

```python
In [5]: len(db('linear_motor'))
Out[5]: 4
```

### CODE REVIEW

**Comment by danielballan** [comment]
^ So, to summarize, this adds a positional argument, `text_search` to `Broker.__call__`, which until now only accepted keyword arguments.

**Comment by licode** [comment]
Good. 
So If I run ```DataBroker('linear_motor')  ``` and ```DataBroker(data_key='linear_motor') ```
this first one will be much faster?

File: databroker/broker.py [hunk 1]
```diff
@@ -183,6 +183,8 @@ def __call__(self, **kwargs):
 
         Parameters
         ----------
+        text_serach : str, optional
```
Comment (klauer) [inline databroker/broker.py]: `text_se ar ch`

File: databroker/broker.py [hunk 2]
```diff
@@ -241,10 +243,16 @@ def __call__(self, **kwargs):
...
                     result.append(rs)
             run_start = result
-        result = []
+        elif text_search is not None:
+            query = {'$text': {'$search': text_search}}
+            run_start = self.mds.find_run_starts(**query)
+        else:
+            query = kwargs
+            run_start = self.mds.find_run_starts(**query)
+        headers = []
```
Comment (klauer) [inline databroker/broker.py]: (bikeshed) while you're here, this is ripe for a list comprehension

**Comment by klauer** [comment]
I missed the mds PR, so I'll ask here: how does case (in)sensitivity work with these searches?

**Comment by danielballan** [comment]
It's highly configurable -- you can even match special characters like umlaut ~ u -- but it is case insensitive by default I believe.

Sent from Outlook Mobile<https://aka.ms/qtex0l>




On Thu, Apr 28, 2016 at 3:05 PM -0700, "K Lauer" <notifications@github.com<mailto:notifications@github.com>> wrote:


I missed the mds PR, so I'll ask here: how does case (in)sensitivity work with these searches?

**Comment by danielballan** [comment]
I am abandoning this PR. Perhaps some courageous soul will figure out how to configure Travis's mongo to allow text indexes. In the meantime, it is possible to use the text-search feature with this PR by passing a raw mongo query to the broker: `db({'$text': {$'search': keyword}})`

**Pull request merged**